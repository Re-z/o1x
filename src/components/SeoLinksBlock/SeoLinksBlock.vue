<template>
    <div class="flex justify-center bg-gray-300 px-4 py-2">
        <div class="mr-5 w-1/2 max-w-[600px]">
            <img
                class="mx-auto mb-4"
                src="https://place-hold.it/100x100"
                alt=""
            />
            <div class="flex flex-wrap items-center justify-center">
                <strong class="mr-4">{{ $t('sections') }}:</strong>
                <Button
                    :type="BUTTON_TYPES.SECONDARY"
                    :color="BUTTON_COLORS.BLACK"
                    v-for="link in links.sections"
                    :key="link.name"
                    :text="link.name"
                    class="mr-2"
                />
            </div>
        </div>
        <div class="w-1/2 max-w-[600px]">
            <img
                class="mx-auto mb-4"
                src="https://place-hold.it/100x100"
                alt=""
            />

            <div class="flex flex-wrap items-center justify-center">
                <strong class="mr-4">{{ $t('popularRequests') }}:</strong>
                <Button
                    :type="BUTTON_TYPES.SECONDARY"
                    :color="BUTTON_COLORS.BLACK"
                    v-for="link in links.popularRequests"
                    :key="link.name"
                    :text="link.name"
                    class="mr-2"
                />
            </div>
        </div>
    </div>
</template>
<script setup lang="ts">
    // imports
    import { useSeoLinksStore } from '@/stores/seoLinks'
    import Button from '@/components/generic/Button/Button.vue'
    import { BUTTON_COLORS, BUTTON_TYPES } from '@/components/generic/Button/Button'

    const links = useSeoLinksStore()
</script>
