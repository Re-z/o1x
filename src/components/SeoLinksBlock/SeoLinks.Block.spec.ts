import { describe, test, expect } from 'vitest'
import { mountComponent } from '@/__tests__/setup'
import SeoLinksBlock from '@/components/SeoLinksBlock/SeoLinksBlock.vue'

describe('SeoLinksBlock.vue tests', () => {
    test('snapshot tests', async () => {
        const wrapper = mountComponent(SeoLinksBlock)
        // snapshot
        expect(wrapper.html()).toMatchSnapshot()
    })
})
