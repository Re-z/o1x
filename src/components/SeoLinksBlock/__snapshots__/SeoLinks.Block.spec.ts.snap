// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SeoLinksBlock.vue tests > snapshot tests 1`] = `
"<div class="flex justify-center bg-gray-300 px-4 py-2">
  <div class="mr-5 w-1/2 max-w-[600px]"><img class="mx-auto mb-4" src="https://place-hold.it/100x100" alt="">
    <div class="flex flex-wrap items-center justify-center"><strong class="mr-4">Розділи:</strong><!-- TODO: optimize -->
      <div class="button-wrapper mr-2">
        <!--v-if-->
        <div class="transform cursor-pointer text-lg transition duration-500 ease-in-out hover:underline active:scale-95" style="color: rgb(0, 0, 0);">Section 1</div>
      </div><!-- TODO: optimize -->
      <div class="button-wrapper mr-2">
        <!--v-if-->
        <div class="transform cursor-pointer text-lg transition duration-500 ease-in-out hover:underline active:scale-95" style="color: rgb(0, 0, 0);">Section 2</div>
      </div><!-- TODO: optimize -->
      <div class="button-wrapper mr-2">
        <!--v-if-->
        <div class="transform cursor-pointer text-lg transition duration-500 ease-in-out hover:underline active:scale-95" style="color: rgb(0, 0, 0);">Section 3</div>
      </div>
    </div>
  </div>
  <div class="w-1/2 max-w-[600px]"><img class="mx-auto mb-4" src="https://place-hold.it/100x100" alt="">
    <div class="flex flex-wrap items-center justify-center"><strong class="mr-4">Популярні запити:</strong><!-- TODO: optimize -->
      <div class="button-wrapper mr-2">
        <!--v-if-->
        <div class="transform cursor-pointer text-lg transition duration-500 ease-in-out hover:underline active:scale-95" style="color: rgb(0, 0, 0);">Request 1</div>
      </div><!-- TODO: optimize -->
      <div class="button-wrapper mr-2">
        <!--v-if-->
        <div class="transform cursor-pointer text-lg transition duration-500 ease-in-out hover:underline active:scale-95" style="color: rgb(0, 0, 0);">Request 2</div>
      </div><!-- TODO: optimize -->
      <div class="button-wrapper mr-2">
        <!--v-if-->
        <div class="transform cursor-pointer text-lg transition duration-500 ease-in-out hover:underline active:scale-95" style="color: rgb(0, 0, 0);">Request 3</div>
      </div>
    </div>
  </div>
</div>"
`;
